Missing access token for scope:funcx_service

---
name: Bug report
about: Create a report to help us improve
title: ''
labels: bug
assignees: ''

---

**Describe the bug**
Having scope:funcx_service not in the login flow.
```
File "/app/diamond_backend/app/utils/login_flow.py", line 77, in initialize_compute_login_manager
    assert funcx_service_token, "Missing access token for scope:funcx_service"
           ^^^^^^^^^^^^^^^^^^^
AssertionError: Missing access token for scope:funcx_service
```
Scope is hardcoded [here](https://github.com/Diamond-Proj/diamond-admin-dashboard/blob/main/src/lib/auth/auth.ts#L33)

It should ideally be like [this](https://github.com/Diamond-Proj/diamond-admin-dashboard/blob/main/src/components/login-button.tsx#L19) - taking from env vars 

See comment for details. https://github.com/Diamond-Proj/diamond-admin-backend/pull/128#issuecomment-3642889841 

**To Reproduce**
Steps to reproduce the behavior:
1. Go to diamondhpc.ai
2. Click on Tasks tab - /tasks page
3. See error

**Expected behavior**
Users should be able to see the list of tasks. Funcx_scope should be fetched from env vars.

--- a/src/lib/auth/auth.ts
+++ b/src/lib/auth/auth.ts
@@ -20,6 +20,7 @@ function redirectToSignIn(request: NextRequest): NextResponse {
 function initiateGlobusLogin(request: NextRequest): NextResponse {
   try {
     const clientId = process.env.NEXT_PUBLIC_GLOBUS_CLIENT_ID;
+    const scopes = process.env.NEXT_PUBLIC_GLOBUS_SCOPES;
     const redirectUri = `${getBaseUrl(request)}/auth/callback`;
 
     if (!clientId) {
@@ -30,8 +31,7 @@ function initiateGlobusLogin(request: NextRequest): NextResponse {
       client_id: clientId,
       redirect_uri: redirectUri,
       response_type: 'code',
-      scope:
-        'openid email profile urn:globus:auth:scope:transfer.api.globus.org:all',
+      scope: scopes || 'openid email profile urn:globus:auth:scope:transfer.api.globus.org:all https://auth.globus.org/scopes/facd7ccc-c5f4-42aa-916b-a0e270e2c2a9/all',
       access_type: 'offline'
     });
 

